crypto/crypto/src/hash.rs

```diff
@@ -605,3 +616,101 @@ impl<T: ser::Serialize + ?Sized> TestOnlyHash for T {
         hasher.finish()
     }
 }
+
+/// Alternative to CryptoHash
+pub trait SimpleHash {
+    /// The hash method
+    fn hash(&self) -> HashValue;
+
+    #[cfg(any(test, feature = "fuzzing"))]
+    /// The salt used for hashing this type.
+    fn salt() -> Vec<u8>;
```

Comment:
The salt should probably be something that is unchanging based on internal state, right? So this should instead have a signature of:

```suggestion
    fn salt() -> &'static [u8];
```

Reply:
We would have to memoize it as well then.